libs/coin-modules/coin-icon/src/logic.ts

```diff
@@ -56,8 +56,8 @@ export const getNonce = (account: IconAccount): number => {
 
   const nonce = Math.max(
     account.iconResources?.nonce || 0,
-    lastPendingOp && typeof lastPendingOp.transactionSequenceNumber === "number"
-      ? lastPendingOp.transactionSequenceNumber + 1
+    lastPendingOp && BigNumber.isBigNumber(lastPendingOp.transactionSequenceNumber)
+      ? Number(lastPendingOp.transactionSequenceNumber.plus(1))
```

Comment:
Not sure if it's better but I would call the `.toNumber()` method of BigNumber.js
```suggestion
      ? lastPendingOp.transactionSequenceNumber.plus(1).toNumber()
```